src/main/java/baritone/pathing/movement/MovementHelper.java

```diff
@@ -280,6 +281,45 @@ static boolean avoidWalkingInto(Block block) {
                 || block == Blocks.WEB;
     }
 
+    static boolean isOverMagma(IPlayerContext ctx, BetterBlockPos src, BetterBlockPos dest) {
+        int x = ctx.playerFeet().x;
+        int y = ctx.playerFeet().y;
+        int z = ctx.playerFeet().z;
+        BlockStateInterface bsi = new BlockStateInterface(ctx);
+
+        if (hasFrostWalker(ctx)) { //if we are wearing frostwalker boots magma can go fuck itself
+            return false;
+        }
+        if (isOverMagma(bsi, x, y, z) || isOverMagma(bsi, dest.x, y, dest.z)) { //we are or going to stand on magma
+            return true;
+        }
+        if (isOverMagma(bsi, dest.x, dest.y, dest.z) && y < src.y) {            //we drop down on a magma block
+            return true;
+        }
+        if ((src.x != dest.x && src.z != dest.z)) {                             //we move diagonaly
+            return isOverMagma(bsi, src.x, y, dest.z) || isOverMagma(bsi, dest.x, y, dest.z) || isOverMagma(bsi, dest.x, y, src.z); //over a magmablock
+        }
+        return false; //if we get this far there is no magma
+    }
+    static boolean isOverMagma(BlockStateInterface bsi, int x, int y, int z) {
+        return bsi.get0(x,y-1,z).getBlock() == Blocks.MAGMA;
+    }
+
+    static boolean hasFrostWalker(CalculationContext context) {
+        return hasFrostWalker(context.getBaritone().getPlayerContext());
+    }
```

Comment:
Perhaps I should have named reasons so you know why I called this a no-go.
The fact whether nor not the player has frost walker is constant during a pathing operation so the suggestion was to cache the boolean result of `EnchantmentHelper.hasFrostWalkerEnchantment` (what I had in mind is `public final boolean frostWalker = EnchantmentHelper.hasFrostWalkerEnchantment(player);`).
There's a rule about cost calculation doing no object allocation and `EnchantmentHelper` breaks that rule.
The pathing thread may not access game state because Minecraft is singlethreaded.
Recalculating a constant two and a half million times per second is a performance nightmare.